### PR TITLE
SAK-48629 assignments: block reads on draft search results

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-48629

### CODE REVIEW

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentContentProducer.java [hunk 1]
```diff
@@ -183,12 +183,31 @@ public Iterator<String> getSiteContentIterator(String context) {
     }
 
     public boolean isForIndex(String ref) {
+
         return ref.startsWith("/assignment");
```
Comment (bjones86) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentContentProducer.java:187]: AssignmentServiceConstants.REFERENCE_ROOT

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentContentProducer.java [hunk 2]
```diff
@@ -183,12 +183,31 @@ public Iterator<String> getSiteContentIterator(String context) {
...
     }
 
     public boolean canRead(String ref) {
-        return assignmentService.permissionCheck(AssignmentServiceConstants.SECURE_ACCESS_ASSIGNMENT, ref, null);
-        //return getAssignment(AssignmentReferenceReckoner.reckoner().reference(ref).reckon()).isPresent();
+
+        if (!ref.startsWith("/assignment")) {
```
Comment (bjones86) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentContentProducer.java:192]:
```suggestion
        if (!isForIndex(ref)) {
```

Comment (jonespm) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentContentProducer.java:192]: Should swap these with `AssignmentServiceConstants.REFERENCE_ROOT`

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentContentProducer.java (hunk 1 above)
Comment (bjones86) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentContentProducer.java:187]:
```suggestion
        return ref.startsWith(AssignmentServiceConstants.REFERENCE_ROOT);
```

**Review by ern — APPROVED** [review]

**Pull request merged**